package.json

```diff
@@ -2,13 +2,26 @@
   "name": "ng-redux",
   "version": "3.4.1",
   "description": "Redux bindings for Angular.js",
-  "main": "./lib/index.js",
+  "browser": "dist/ng-redux.js",
+  "main": "lib/ng-redux.js",
+  "module": "es/ng-redux.js",
+  "typings": "./index.d.ts",
+  "files": [
+    "dist",
+    "lib",
+    "es",
+    "src",
+    "index.d.ts"
+  ],
   "scripts": {
-    "build": "npm run build:lib && npm run build:min && npm run build:max",
-    "build:lib": "rm -rf lib && babel src --out-dir lib",
-    "build:min": "webpack --optimize-minimize --define process.env.NODE_ENV=\"'production'\" lib/index.js dist/ng-redux.min.js",
-    "build:max": "webpack lib/index.js dist/ng-redux.js",
-    "test": "mocha --compilers js:babel-register --recursive"
+    "clean": "rimraf lib dist es",
+    "build:commonjs": "cross-env NODE_ENV=cjs rollup -c -o lib/ng-redux.js",
+    "build:es": "cross-env NODE_ENV=es rollup -c -o es/ng-redux.js",
+    "build:umd": "cross-env NODE_ENV=development rollup -c -o dist/ng-redux.js",
+    "build:umd:min": "cross-env NODE_ENV=production rollup -c -o dist/ng-redux.min.js",
+    "build": "yarn run build:commonjs && yarn run build:es && yarn run build:umd && yarn run build:umd:min",
+    "test": "NODE_ENV=test mocha --compilers js:babel-register --recursive",
+    "prepublish": "yarn run clean && yarn test && yarn run build"
```

Comment:
@deini why did we switch to using `yarn` over `npm` in the build process? We don't have `yarn` in the dependencies, and I, for example, don't have yarn on my machine because it's still screwy in some ways and not as cross-system compatible (Windows definitely suffers here).